template/ios/Podfile

```diff
@@ -0,0 +1,45 @@
+platform :ios, '9.0'
+
+target 'HelloWorld' do
+  # Pods for HelloWorld
+  pod 'React', :path => '../node_modules/react-native/'
+  pod 'React-Core', :path => '../node_modules/react-native/React'
+  pod 'React-DevSupport', :path => '../node_modules/react-native/React'
+  pod 'React-fishhook', :path => '../node_modules/react-native/Libraries/fishhook'
+  pod 'React-RCTActionSheet', :path => '../node_modules/react-native/Libraries/ActionSheetIOS'
+  pod 'React-RCTAnimation', :path => '../node_modules/react-native/Libraries/NativeAnimation'
+  pod 'React-RCTBlob', :path => '../node_modules/react-native/Libraries/Blob'
+  pod 'React-RCTGeolocation', :path => '../node_modules/react-native/Libraries/Geolocation'
+  pod 'React-RCTImage', :path => '../node_modules/react-native/Libraries/Image'
+  pod 'React-RCTLinking', :path => '../node_modules/react-native/Libraries/LinkingIOS'
+  pod 'React-RCTNetwork', :path => '../node_modules/react-native/Libraries/Network'
+  pod 'React-RCTSettings', :path => '../node_modules/react-native/Libraries/Settings'
+  pod 'React-RCTText', :path => '../node_modules/react-native/Libraries/Text'
+  pod 'React-RCTVibration', :path => '../node_modules/react-native/Libraries/Vibration'
+  pod 'React-RCTWebSocket', :path => '../node_modules/react-native/Libraries/WebSocket'
+
+  pod 'React-cxxreact', :path => '../node_modules/react-native/ReactCommon/cxxreact'
+  pod 'React-jsi', :path => '../node_modules/react-native/ReactCommon/jsi'
+  pod 'React-jsiexecutor', :path => '../node_modules/react-native/ReactCommon/jsiexecutor'
+  pod 'React-jsinspector', :path => '../node_modules/react-native/ReactCommon/jsinspector'
+  pod 'yoga', :path => '../node_modules/react-native/ReactCommon/yoga'
```

Comment:
Not a blocker, but all of the above could be grabbed via a glob for Podspecs:

```ruby
Dir.glob('../node_modules/react-native/**/*.podspec').each do |react_pod| 
  name = File.basename(react_pod, "podspec")
  path = File.dirname(react_pod)
  pod name, :path => path
end
```

This'll probably get deprecated by https://github.com/react-native-community/react-native-cli/pull/256 eventually though

Reply:
This is so cool we can handle built-in podspecs with that API too.

Reply:
@orta this is cool and much nicer for version upgrades. However, the code above doesn't produce the same result, because not all podspecs in the code base are included by default. These libraries were not linked in by default in the old Xcode project and correspondingly those pods are not dependencies of the `React` podspec:
```
./Libraries/ART/React-ART.podspec
./Libraries/CameraRoll/React-RCTCameraRoll.podspec
./Libraries/PushNotificationIOS/React-RCTPushNotification.podspec
./React/React-RCTFabric.podspec
./ReactCommon/fabric/graphics/React-graphics.podspec
./ReactCommon/React-Fabric.podspec
./ReactCommon/turbomodule/core/React-turbomodule-core.podspec
```

Reply:
I wonder if we make a way keep track of the default Podspecs inside React Natives codebase?

Reply:
Right now it's these podspecs: https://github.com/facebook/react-native/blob/bbd98d5c46a37ebe23ce0424aff1d8c734c8b444/React.podspec#L45-L57 and their transitive dependencies.

Right now the podspec structure is supposed to mirror the Xcode projects as closely as possible, so I put podspecs next to the corresponding Xcode project, but perhaps we could have a designated folder for all "core" podspecs that should be picked up by default?

Reply:
Interesting thanks, I wonder if we can use that then - it's a great source of truth.

Could have the podfile read the React.podspec, and add those dependencies (might end up being a bit of code (because the file paths are non-deterministic) )

I'll have a think